tests/integration.rs

```diff
@@ -332,28 +332,15 @@ fn wildcard_subject_alternative_names() {
 }
 
 #[cfg(feature = "alloc")]
-fn expect_cert_dns_names(cert_der: &[u8], expected_names: &[&str]) {
-    use std::collections::HashSet;
-
+fn expect_cert_dns_names<'name>(
+    cert_der: &[u8],
+    expected_names: impl IntoIterator<Item = &'name str>,
+) {
     let der = CertificateDer::from(cert_der);
     let cert = webpki::EndEntityCert::try_from(&der)
         .expect("should parse end entity certificate correctly");
 
-    let expected_names: HashSet<_> = expected_names.iter().cloned().collect();
-
-    let mut actual_names = cert
-        .dns_names()
-        .expect("should get all DNS names correctly for end entity cert")
-        .collect::<Vec<_>>();
-
-    // Ensure that converting the list to a set doesn't throw away
-    // any duplicates that aren't supposed to be there
-    assert_eq!(actual_names.len(), expected_names.len());
-
-    let actual_names: std::collections::HashSet<&str> =
-        actual_names.drain(..).map(|name| name.into()).collect();
-
-    assert_eq!(actual_names, expected_names);
+    assert!(cert.dns_names().unwrap().eq(expected_names))
```

Comment:
is there a magic that means `assert_eq!(cert.dns_names().unwrap(), expected_names);` didn't work? assert_eq is nice because it prints the two arguments if the assertion fails.

Reply:
I tried using `assert_eq!` initially but:

> binary operation `==` cannot be applied to type `impl Iterator<Item = &str>`
> impl Iterator<Item = &str>` doesn't implement `Debug
> impl IntoIterator<Item = &'name str>` doesn't implement `Debug

The latter two complaints also mean we can't do `assert!(dbg!(cert.dns_names()).eq(dbg!(expected_names)))` :-(